### PR TITLE
Add autosuspend rules for emulated QEMU devices

### PR DESCRIPTION
**Pull request by keszybz**
Partially reverts "rules: remove all power management from udev" /
e2452eef02a839e1928f4ffd893c93a460474ab6.

The rules for emulated QEMU hardware are restored. It seems that they
were removed in one fell swoop with other rules which were causing problems,
despite the qemu rules working properly (and being adjusted through patches
over time).

https://github.com/systemd/systemd/pull/353#issuecomment-658810289

### CODE REVIEW

File: rules.d/60-autosuspend.rules [hunk 1]
```diff
@@ -6,6 +6,18 @@ ACTION!="add", GOTO="autosuspend_end"
...
+#
+# Note that there are buggy qemu versions (0.13 & older) which
+# advertise remote wakeup support but don't actually implement
+# it correctly.  This is the reason why we need a match for the
+# serial number here.  Old, broken versions have serial "1".
+# It has been changed to "42" after fixing the bug to indicate
+# remote wakeup is working.
+SUBSYSTEM=="usb", ATTR{product}=="QEMU USB Mouse", ATTR{serial}!="1", TEST=="power/control", ATTR{power/control}="auto"
+SUBSYSTEM=="usb", ATTR{product}=="QEMU USB Tablet", ATTR{serial}!="1", TEST=="power/control", ATTR{power/control}="auto"
+SUBSYSTEM=="usb", ATTR{product}=="QEMU USB Keyboard", ATTR{serial}!="1", TEST=="power/control", ATTR{power/control}="auto"
```
Comment (poettering) [inline rules.d/60-autosuspend.rules]: I think it would be better to se ID_AUTOSUSPEND here, and let the rule below then set the attribute

**Comment by keszybz** [comment]
Done.

File: src/udev/udev-builtin-hwdb.c [hunk 2]
```diff
@@ -58,15 +58,16 @@ static const char *modalias_usb(sd_device *dev, char *s, size_t size) {
                 return NULL;
         if (safe_atoux16(p, &pn) < 0)
                 return NULL;
-        snprintf(s, size, "usb:v%04Xp%04X*", vn, pn);
+        (void) sd_device_get_sysattr_value(dev, "product", &n);
+
+        snprintf(s, size, "usb:v%04Xp%04X:%s", vn, pn, n);
```
Comment (yuwata) [inline src/udev/udev-builtin-hwdb.c]: Maybe `strna(n)`? Or
```
snprintf(s, size, "usb:v%04Xp%04X%s%s", vn, pn, n ? ":" : "", strempty(n));
```

Comment (keszybz) [reply src/udev/udev-builtin-hwdb.c]: Good point. Updated with `strempty`.

Comment (keszybz) [reply src/udev/udev-builtin-hwdb.c]: (I kept the ":". I think it should be there to allow easy field separation.)

**Comment by yuwata** [comment]
LGTM.

**Comment by poettering** [comment]
excellent!

**Pull request merged**